### ISSUE TITLE
SendMessageAndWaitForResponse <=> Client Disconnect

### ISSUE THREAD

**Issue by zs2hx** [hikalkan/scs#10]
When a client disconnects whilst waiting in SendMessageAndWaitForResponse then:
1. the client Disconnected event gets fired successfully
2. the messenger continues to wait until the timeout is reached.

Can we bubble up the Disconnected event in SendMessageAndWaitForResponse ?

**Comment by hikalkan** [comment]
Good point. I've no time for now, but I can accept a pull request if you implement.
Thanks.
